From 43c18625916da8ed9d6f41ada62122892f2d8297 Mon Sep 17 00:00:00 2001
From: "Lin.TommySC" <lin.tommysc@inventec.com>
Date: Mon, 26 Jul 2021 01:04:32 +0000
Subject: [PATCH] Fix type mismatch when calling sendTrap

Symptom/Reason :
    - Add IPv6 Header Traffic Class support

Root Cause:
    - N/A

Solution/Change:
    [phosphor-ipmi-host]
        - Add handler for IPv6TrafficClass command
    [phosphor-ipmi-config]
        - Add ipv6_traffic_class config field
    [phosphor-snmp]
        - Set IPv6 Header Traffic Class before sending SNMP trap
    [phosphor-dbus-monitor]
        - Fix type mismatch when calling sendTrap

Entry Test:
    - Set IPv6 Header Traffic Class
        ipmitool raw 0x0c 0x01 0x01 0x34 0x2
    - Get IPv6 Header Traffic Class
        ipmitool raw 0x0c 0x02 0x01 0x34 0x00 0x00
---
 src/snmp_trap.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/snmp_trap.cpp b/src/snmp_trap.cpp
index 309b22e..4617d40 100644
--- a/src/snmp_trap.cpp
+++ b/src/snmp_trap.cpp
@@ -36,7 +36,7 @@ void ErrorTrap::trap(sdbusplus::message::message& msg) const
     auto errorID = std::get<uint32_t>(propMap.at("Id"));
     auto timestamp = std::get<uint64_t>(propMap.at("Timestamp"));
     auto sev = std::get<std::string>(propMap.at("Severity"));
-    auto isev = static_cast<uint8_t>(Entry::convertLevelFromString(sev));
+    auto isev = static_cast<int32_t>(Entry::convertLevelFromString(sev));
     auto message = std::get<std::string>(propMap.at("Message"));
     try
     {
